ng-dev/ts-circular-dependencies/index.ts

```diff
@@ -119,6 +107,33 @@ export function main(
     Log.warn(`   Please rerun with "--warnings" to inspect unresolved imports.`);
   }
 
+  if (goldenFile === undefined) {
+    if (approve) {
+      Log.error(
+        `x  Cannot approve circular depdencies within this repository as no golden file exists.`,
+      );
+      return 1;
+    }
+    if (cycles.length > 0) {
```

Comment:
Shouldn't this be outside? What if there is a `goldenFile` and cycles?

Reply:
This is within the `goldenFile` being `undefined` so there is no golden file.

Reply:
But shouldn't this fail? if there are cycles and `goldenFile` being `!== undefined`?

Reply:
I guess it will happen automatically below with the comparison. That's the thing I missed. Very confusing :D

Reply:
For the case where there is no golden file, and there are cycles here. This fails as it returns a `1` as the exit code.

For cases where there is a golden file to do the comparison to later, we check the specifics.